evaluator/src/main/resources/osadl/osadl.rules.kts

```diff
@@ -0,0 +1,112 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+// Use this rule like:
+//
+// $ ort evaluate -i scanner/src/funTest/assets/dummy-expected-output-for-analyzer-result.yml --rules-resource /osadl/osadl.rules.kts
+
+enum class Compatibility {
+    INHERENT,
+    YES,
+    NO,
+    UNKNOWN;
+
+    companion object {
+        val COMPLIANT = enumSetOf(Compatibility.INHERENT, Compatibility.YES)
+
+        fun fromString(value: String): Compatibility =
+            when (value) {
+                "" -> INHERENT
+                "Yes" -> YES
+                "No" -> NO
+                "?", "Dep." -> UNKNOWN
+                else -> throw IllegalArgumentException("Unhandled compatibility '$value' found.")
+            }
+    }
+}
+
+val matrix = mutableMapOf<Pair<String, String>, Compatibility>()
+
+javaClass.getResourceAsStream("/osadl/osadl-matrix.csv").bufferedReader().useLines { lines ->
+    val rowIterator = lines.iterator()
+
+    // The first row contains the header with names of the inbound licenses.
+    val columnHeaders = rowIterator.next().split(',')
+        // Drop the surrounding "Compatibility*" cells from the header row.
+        .drop(1).dropLast(1)
+
+    while (rowIterator.hasNext()) {
+        val cellIterator = rowIterator.next().split(',').iterator()
+
+        // The first column contains the header with names of the outbound licenses.
+        val inbound = cellIterator.next()
+
+        columnHeaders.forEach { outbound ->
+            val value = cellIterator.next()
+            matrix[inbound to outbound] = Compatibility.fromString(value)
+        }
+    }
+}
+
+fun PackageRule.LicenseRule.isOsadlMatrixCompliant(inboundLicenses: Collection<String>) =
+    object : RuleMatcher {
+        override val description = "isOsadlMatrixCompliant($license)"
+
+        override fun matches(): Boolean {
+            val outbound = license.toString()
+            return inboundLicenses.all { matrix[it to outbound] in Compatibility.COMPLIANT }
+        }
+    }
+
+val ruleSet = ruleSet(ortResult, licenseInfoResolver) {
+    // Define a rule that is executed for each dependency.
+    dependencyRule("OSADL compliance matrix") {
+        // Requirements for the rule to trigger a violation.
+        require {
+            -isExcluded()
+        }
+
+        // Define a rule that is executed for each license of the dependency.
+        val licenseView = LicenseView.CONCLUDED_OR_DECLARED_AND_DETECTED
+        licenseRule("OSADL compliance matrix", licenseView) {
+            // Use all licenses of all direct dependencies as inbound licenses.
```

Comment:
Just wondering, why is it sufficient to check only direct dependencies here?

Reply:
Because for each dependency its directing dependencies are checked, so the whole tree is covered in the end. I believe that should be fine, or?

Reply:
I haven't looked at the matrix, but if there are situations where license A is compatible with B, B is compatible with C, but A is not compatible with C, it would not be fine.

Reply:
I've changed this now according to the recommendations of @LeChasseur in the "osadl.rules: Only use the project's license as the outbound license" commit.